.github/workflows/push-pull.yml

```diff
@@ -0,0 +1,20 @@
+name: Push-Pull
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  pyiron:
+    uses: pyiron/actions/.github/workflows/push-pull.yml@pyiron_base_lessons
```

Comment:
Is there an option to test a new python version? like 3.13 ?

Reply:
Yes, of course. One can always use the `unit-tests` action individually in a separate workflow to get a new tests with minimal lines, but the reusable `push-pull` workflow already accommodates running the "main" version of pyiron (default 3.11) across up to three runners (default linux, windows, and macos) and the "main" runner (default linux) across up to three versions of python (default 3.10, 3.9, 3.12). I say "up to" because you can always disable one of these by setting it to "exclude".

Until recently it supported only two alternate versions of python. I am not aware of nice loop-making tools in github's yaml syntax, so new entries need to be manually added. If it becomes helpful to have more runners or python versions covered by default in the reusable workflow, it's an easy extension and I have no objection to adding them for you.

I confess this is not clearly explained at a high level, but all this knowledge is available in the workflow argument descriptions:
https://github.com/pyiron/actions/blob/4bac4483cd063937b27871f5eade0e68a546c7ed/.github/workflows/push-pull.yml#L88-L122